client/lib/wpcom-undocumented/lib/undocumented.js

```diff
@@ -288,7 +288,7 @@ Undocumented.prototype.settings = function( siteId, method, data, fn ) {
 		data = {};
 	}
 
-	this.wpcom.req[ method ]( {
+	return this.wpcom.req[ method ]( {
```

Comment:
This can create a bug if you try to use it like a promise because (method undefined). Maybe add default values.

I had this issue on my PR as well, so maybe worth checking the other endpoints.